rust/arrow/src/array/transform/mod.rs

```diff
@@ -298,79 +290,137 @@ impl<'a> MutableArrayData<'a> {
             use_nulls = true;
         };
 
-        let buffers = match &data_type {
+        let empty_buffer = MutableBuffer::new(0);
+        let buffers: [MutableBuffer; 2] = match &data_type {
```

Comment:
Very minor suggestion:
```suggestion
        let (buffer1, buffer2) = match &data_type {
```
And you can remove the destructuring below.